### PR TITLE
Closes #10617: Add common domain for icon requests

### LINKED ISSUE #10617: Simplify Icon Requests for Wikipedia
Currently URL of different Wikipedia language sites are considered different.  (Ex: en.m.wikipedia.org, en.wikipedia.org, de.wikipedia.org etc) when retrieving icons.  However, they should all be considered as wikipedia.org.

Add a common domain list so if the URL ends with it we reduce it to the common domain.  For example:
en.wikipedia.org -> wikipedia.org
de.m.wikipedia.org - -> wikipedia.org

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/android/net/Uri.kt [hunk 1]
```diff
@@ -17,6 +17,19 @@ import java.io.IOException
...
+ * Returns the host's common domain if found, else null is returned
+ */
+val Uri.hostWithCommonDomain: String?
+    get() {
+        val host = host ?: return null
+        for (domain in commonDomain) {
+            if (host.endsWith(domain)) return domain
+        }
+        return null
+    }
```
Comment (pocmo) [inline components/support/ktx/src/main/java/mozilla/components/support/ktx/android/net/Uri.kt:32]: I would consider moving this code inside `TippyTopIconPreparer`. It is very specific to that use case and outside of it "common domain" is not really a concept that makes sense. This is specifically crafted for the tippy top sites list :)

**Comment by Amejia481** [comment]
There is also a test failure  `getRepresentativeCharacter` :)

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/android/net/Uri.kt (hunk 1 above)
Comment (rocketsroger) [reply components/support/ktx/src/main/java/mozilla/components/support/ktx/android/net/Uri.kt:32]: Sure, I'll do that.  Thanks

File: components/browser/icons/src/main/java/mozilla/components/browser/icons/preparer/TippyTopIconPreparer.kt [hunk 2]
```diff
@@ -18,6 +19,21 @@ import org.json.JSONObject
 
 private const val LIST_FILE_PATH = "mozac.browser.icons/icons-top200.json"
 
+// Make sure domain added here have the corresponding image_url in icons-top200.json
+private val commonDomain = listOf("wikipedia.org")
+
+/**
+ * Returns the host's common domain if found, else null is returned
+ */
+val Uri.hostWithCommonDomain: String?
```
Comment (pocmo) [inline components/browser/icons/src/main/java/mozilla/components/browser/icons/preparer/TippyTopIconPreparer.kt:28]: This is still public to everyone :)

Comment (rocketsroger) [reply components/browser/icons/src/main/java/mozilla/components/browser/icons/preparer/TippyTopIconPreparer.kt:28]: oh that's true, I'll make it private

**Review by pocmo — APPROVED** [review]

**Pull request merged**